zcash_client_sqlite/src/wallet/init.rs

```diff
@@ -513,19 +517,20 @@ mod tests {
                    blocks.time                       AS block_time,
                    (
                         blocks.height IS NULL
-                        AND transactions.expiry_height <= blocks_max_height.max_height
+                        AND transactions.expiry_height BETWEEN 1 AND blocks_max_height.max_height
                    ) AS expired_unmined
-            FROM transactions
-            JOIN notes ON notes.id_tx = transactions.id_tx
+            FROM notes
+            LEFT OUTER JOIN transactions
+                 ON notes.txid = transactions.txid
             JOIN blocks_max_height
             LEFT JOIN blocks ON blocks.height = transactions.block
```

Comment:
This join is still being done between `blocks` and `transactions`, which means that for UTXOs, even if we have scanned the block the UTXO was received in, we will report it as having `block_time = null` (which ends up being set to 0 somewhere in some of the downstream SDK usages). If we join `blocks` to `notes` instead, we avoid this problem.
```suggestion
            LEFT JOIN blocks ON blocks.height = notes.block
```